exporter/otlpexporter/internal/arrow/stream.go

```diff
@@ -224,22 +224,25 @@ func (s *Stream) SendAndWait(ctx context.Context, records interface{}) error {
 }
 
 // encode produces the next batch of Arrow records.
-func (s *Stream) encode(records interface{}) (*arrowpb.BatchArrowRecords, error) {
+func (s *Stream) encode(records interface{}) (_ *arrowpb.BatchArrowRecords, retErr error) {
 	// Note!! This is a placeholder.  After this PR merges the
 	// code base will be upgraded to the latest version of
 	// github.com/f5/otel-arrow-adapter, which returns one
 	// BatchArrowRecords per pdata Logs/Metrics/Traces.  The TODO below
 	// will be addressed, i.e., we will not drop all but one
 	// batch.
+	defer func() {
+		if err := recover(); err != nil {
+			retErr = fmt.Errorf("panic in otel-arrow-adapter: %v", err)
+		}
+	}()
```

Comment:
In case this raises 👀, I had been seeing a panic in an earlier otel-arrow-adapter and this was a defensive move.